feat: [PKD-019][PKD-021][PKD-022][PKD-023] Fix Enhancement App Bar & …

Enrickzz opened the pull request:
…Fix Overflow Issue & Fix Pokemon Details Tab View & Enhancement: Home Screen Tile Spacing

- Updated handler to include abilities, height, weight, baseExp
- Fixed Overflow issue (pokemon_tile)
- Pokemon Tile UI update (colors, type, image)
- HomePage tiles spacing fix
- HomePage AppBar update (removed second scaffold and appbar)
- Added about widget for pokemon_detail_page
- Some UI updates on base_stats_tab and pokemon_detail_page

https://user-images.githubusercontent.com/61441278/181424901-574ff0e4-b927-41c6-9c1c-29ddf40159e7.mp4

Enrickzz committed:
feat: [PKD-019][PKD-021][PKD-022][PKD-023] Fix Enhancement App Bar & Fix Overflow Issue & Fix Pokemon Details Tab View & Enhancement: Home Screen Tile Spacing

- Updated handler to include abilities, height, weight, baseExp
- Fixed Overflow issue (pokemon_tile)
- Pokemon Tile UI update (colors, type, image)
- HomePage tiles spacing fix
- HomePage AppBar update (removed second scaffold and appbar)
- Added about widget for pokemon_detail_page
- Some UI updates on base_stats_tab and pokemon_detail_page

Enrickzz committed:
PR Feedback

- Pokemon Tile Overflow fix
- Separated Tab View Widget
- Details Page Fill fix
- Added Loading widgets (Spin Kit)

dbenedictos commented on lib/api/handlers/details_pokemon_handler.dart:
```diff
@@ -21,10 +23,21 @@ class DetailsPokemonHandler {
+        weight: endResponse['weight'],
+        abilities: abilitiesMap
+            .map((e) => Ability(
+                  ability: SpecificAbility(name: e['ability']['name'], url: e['ability']['url']),
```
trailing comma

dbenedictos replied:
trailing comma

dbenedictos commented on lib/feature/home_page/home_page.dart:
```diff
@@ -23,61 +23,40 @@ class HomePage extends StatelessWidget {
+        },
+        child: Padding(
+          padding: EdgeInsets.fromLTRB(5, 10, 10, 0),
+          child: GridView.builder(
```
use GridView.count

dbenedictos commented on lib/feature/home_page/widgets/pokemon_tile.dart:
```diff
@@ -49,57 +52,62 @@ class _PokemonTileState extends State<PokemonTile> {
-                )
-              ],
+                  SizedBox(height: 5),
+                  Visibility(
```
instead of using visibility. Add an if statement
`if(condition)`
`  PillContainterWidget()`

dbenedictos commented on lib/feature/home_page/widgets/pokemon_tile.dart:
```diff
@@ -49,57 +52,62 @@ class _PokemonTileState extends State<PokemonTile> {
+                '$pokemonImgUrl${thisTileTypes?.id}.png',
+                width: 115,
+                height: 115,
+                errorBuilder: (_, __, ___) {
```
use `loadingBuilder` for your loading widget and make a separate widget that would look good that represents the error, use that in your `errorBuilder`

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,150 +15,75 @@ class PokemonDetails extends StatelessWidget {
+                  },
+                ),
+              )
+            ]),
```
trailing comma

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,150 +15,75 @@ class PokemonDetails extends StatelessWidget {
+                      type: pokemon.types!.first.type!.name!,
+                      color: typeDetailsPageBackgroundColor,
+                    ),
+                    Visibility(
```
same here use an if statement

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,150 +15,75 @@ class PokemonDetails extends StatelessWidget {
+                child: Image.network(
+                  '$pokemonImgUrl${pokemon.id}.png',
+                  fit: BoxFit.cover,
+                  errorBuilder: (_, __, ___) {
```
same comment as above. Use `loadingBuilder`

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+  final int height;
+  final int weight;
+  final List<Ability>? abilities;
+  final int baseExperience;
```
please group similarly typed variables together

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                  child: Container(
+                    width: 150,
+                    child: Text(
+                      'Height',
```
Strings file

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                  child: Container(
+                    width: 150,
+                    child: Text(
+                      'Weight',
```
and this 😛

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                  child: Container(
+                    width: 150,
+                    child: Text(
+                      'Abilities',
```
and this haha

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                Container(
+                  width: 200,
+                  child:
+                      Wrap(children: abilities!.map((e) => Text('${(e.ability?.name ?? '').capitalize}, ')).toList()),
```
null check instead of force unwrap

dbenedictos replied:
this will result in an unwanted comma
![Screenshot 2022-07-28 at 4 51 44 PM](https://user-images.githubusercontent.com/47196695/181464484-b408700b-8cea-4ba9-9c14-bdcf9006460e.png)

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                  child: Container(
+                    width: 150,
+                    child: Text(
+                      'Base Experience ',
```
strings file

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_about_tab.dart:
```diff
@@ -0,0 +1,124 @@
+                ),
+                Container(
+                  width: 60,
+                  child: Text('$baseExperience'),
```
`baseExperience.toString`

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+              topLeft: Radius.circular(30),
+              topRight: Radius.circular(30),
+            ),
+            color: Colors.white),
```
trailing comma

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+                      child: Text(
+                    pokemonDetailsTabs[1],
+                    style: TextStyle(color: Colors.black),
+                  )),
```
trailing comma

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+                      child: Text(
+                    pokemonDetailsTabs[2],
+                    style: TextStyle(color: Colors.black),
+                  )),
```
here

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+                      child: Text(
+                    pokemonDetailsTabs[3],
+                    style: TextStyle(color: Colors.black),
+                  )),
```
and here

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+                  child: TabBarView(
+                children: <Widget>[
+                  AboutTab(
+                    weight: pokemon.weight!,
```
null check and assign a fallback

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+
+class TabView extends StatelessWidget {
+  const TabView({required this.pokemon});
+  final DetailsPokemon pokemon;
```
please rename to `pokemonDetails` or `detailsPokemon`

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,77 @@
+                    abilities: pokemon.abilities,
+                    baseExperience: pokemon.baseExperience!,
+                  ),
+                  BaseStats(stat: pokemon.stats!),
```
stats in DetailsPokemon is nullable, so in your BaseStats widget don't require the parameter

dbenedictos commented on lib/feature/widgets/pill_container_widget.dart:
```diff
@@ -20,7 +20,7 @@ class PillContainerWidget extends StatelessWidget {
           padding: EdgeInsets.fromLTRB(10, 3, 10, 3),
           child: Text(
-            type.capitalize,
+            '${type?.capitalize}',
```
already a string

Enrickzz committed:
PR Feedback

- Convert Visibility Widget to if statements
- Convert Gridview.builder to Gridview.count
- Added strings to strings constants
- Make Stats nullable in BaseStats Widget
- Renamed pokemon => pokemonDetails
- Assigned fallbacks to parameters
- Reworked abilities row (about tab)

dbenedictos commented on lib/feature/home_page/widgets/pokemon_tile.dart:
```diff
@@ -49,57 +52,56 @@ class _PokemonTileState extends State<PokemonTile> {
-                )
-              ],
+                  SizedBox(height: 5),
+                  thisTileTypes?.subTypes?.last.type?.name == thisTileTypes?.subTypes?.first.type?.name
```
'if(thisTileTypes?.subTypes?.last.type?.name == thisTileTypes?.subTypes?.first.type?.name)'
'PillContainerWidget'

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,152 +12,71 @@ class PokemonDetails extends StatelessWidget {
+          ]),
+        ),
+        Expanded(child: TabView(pokemonDetails: pokemon))
+      ]),
```
trailing comma

dbenedictos replied:
same here

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,152 +12,71 @@ class PokemonDetails extends StatelessWidget {
+                errorBuilder: (_, __, ___) => SpinKitSpinningLines(color: Colors.white),
+              ),
+            )
+          ]),
```
trailing comma

dbenedictos replied:
@Enrickzz add a comma between the square bracket and the parenthesis

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_tab_view.dart:
```diff
@@ -0,0 +1,81 @@
+                    color: (pokemonDetails.types?.first.type?.name ?? '').getPokemonColor,
+                  ),
+                ],
+              )),
```
trailing comma

Enrickzz committed:
PR Feedbacks

- Trailing commas
- Revise ifs
- Group similar var types
- Nullable Pill Container

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -13,152 +12,70 @@ class PokemonDetails extends StatelessWidget {
+                    text: pokemon.types!.first.type!.name!,
+                    color: typeDetailsPageBackgroundColor,
+                  ),
+                  if (pokemon.types?.last.type?.name != pokemon.types?.first.type?.name)
```
gonna let this go for now, since we're gonna refactor the Types

dbenedictos commented on lib/feature/widgets/pill_container_widget.dart:
```diff
@@ -3,24 +3,24 @@ import 'package:pokedex_async_redux/utilities/extensions.dart';
   });
-  final String type;
-  final Color color;
+  final String? text;
```
remove the nullable notation since it's required in the constructor

Enrickzz committed:
PR Feedback

- PillContainer remove nullable notation
- Trailiing commas
